plutus-core/plutus-core/src/PlutusCore/Compiler/Types.hs

```diff
@@ -0,0 +1,10 @@
+{-# LANGUAGE ConstraintKinds #-}
+
+module PlutusCore.Compiler.Types where
+
+import PlutusCore.Builtin
+import PlutusCore.Name
+import PlutusCore.Quote
+
+type Compiling m uni fun name =
+  (ToBuiltinMeaning uni fun, MonadQuote m, HasUnique name TermUnique, Ord name)
```

Comment:
`Ord name` is weird, do you know where it comes from?

Reply:
Those are supposed to use `UniqueMap` and we could add `UniqueSet` too. Hard for me to imagine why we'd prefer `Map` over a properly `newtype`d `IntMap`.

Reply:
Good point.